src/EasyCaching.Redis/DefaultRedisCachingProvider.Async.cs

```diff
@@ -295,6 +295,27 @@ public override async Task BaseSetAllAsync<T>(IDictionary<string, T> values, Tim
 
             return result;
         }
+        
+        
+        /// <summary>
+        /// Gets all keys async by prefix.
+        /// </summary>
+        /// <param name="prefix">Prefix</param>
+        /// <param name="cancellationToken">CancellationToken</param>
+        /// <returns>The all keys by prefix async.</returns>
+        public override async Task<IEnumerable<string>> BaseGetAllKeysByPrefixAsync(string prefix, CancellationToken cancellationToken = default)
+        {
+            ArgumentCheck.NotNullOrWhiteSpace(prefix, nameof(prefix));
+
+            prefix = this.HandlePrefix(prefix);
+
+            var redisKeys = this.SearchRedisKeys(prefix);
+            
+            var result = redisKeys?.Select(key => (string) key)?.Distinct();
```

Comment:
Suppose we do not set a global prefix.

We set something as below

```
Set("order:aaaa", "1", TimeSpan.FromDay(1));
Set("order:aaab", "1", TimeSpan.FromDay(1));
Set("order:aaac", "1", TimeSpan.FromDay(1));
```

When we call `GetAllKeysByPrefixAsync("order", default)`,  the result will be  `["order:aaaa", "order:aaab", "order:aaac"]`

This is okey.

But, suppose we do set a global prefix.

```
services.AddEasyCaching(x =>
 {            
                x.UseRedis(options =>
                {
                    options.DBConfig = new RedisDBOptions
                    {
                        KeyPrefix = "order:"
                    };
                   // ....
                }, "WithKeyPrefix");
            });
```

Then, we set something as below

```
Set("order:aaaa", "1", TimeSpan.FromDay(1));
Set("order:aaab", "1", TimeSpan.FromDay(1));
Set("order:aaac", "1", TimeSpan.FromDay(1));
```

When we call `GetAllKeysByPrefixAsync("order", default)`,  which result do you expect?

 `["order:aaaa", "order:aaab", "order:aaac"]` or  `["order:order:aaaa", "order:order:aaab", "order:order:aaac"]`

Reply:
Hi,
I Think right result is best: `["order:order:aaaa", "order:order:aaab", "order:order:aaac"]`. we add `"order"` from `GetAllKeysByPrefixAsync("order", default)` to base prefix that here is `order:` and result `order:  + order` and we have `order:order` in start key prefix. We consider `base prefix first` and combine it with `parameter prefix` from method. If you are agree with this approach I change my code to this solution.

Reply:
When you get this result `["order:order:aaaa", "order:order:aaab", "order:order:aaac"]`, and if you want to get one of the value of them, you will call `Get<T>("order:order:aaaa")`, but it cannot get the value if you set a base/global  preifx, at this time `["order:aaaa", "order:aaab", "order:aaac"]` is a better result.

If you only want to get the cache key store in redis,  `["order:order:aaaa", "order:order:aaab", "order:order:aaac"]` is ok.

If you want to get the cache key and call other methods provided by EasyCaching, such as `Get<T>`,  `["order:aaaa", "order:aaab", "order:aaac"]` is ok.

This is why I suggest you pay attention to deal with the key prefix.

Reply:
I wanna get all cache keys and using this keys in other method EasyCaching like `Get<T>` and for handle this scenario I change my code and now if we call `GetAllKeysByPrefixAsync("order", default)` and have `base key prefix = order:`, we  get this result `["order:aaaa", "order:aaab", "order:aaac"]` as well with calling this method `Get<T>`.

Full changeset is here:
cd5d01f1a1f4e31649507380aec605742fe55b29